esx_service/vmdk_ops.py

```diff
@@ -786,9 +786,23 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
     vm_datastore_url = vmdk_utils.get_datastore_url_from_config_path(config_path)
     vm_datastore = get_datastore_name(vm_datastore_url)
 
+    # Detach is implicity allowed if a disk has been attached
+    # Do not find tenant or privileges; just detach the disk
+    if cmd == "detach":
+        with lockManager.get_lock(vm_uuid):
+            response = detachVMDK(full_vol_name, vm_uuid)
+
+        return response
+
     error_info, tenant_uuid, tenant_name = auth.get_tenant(vm_uuid)
     if error_info:
-        return err(error_info)
+        # For docker volume ls, docker prints a list of cached volume names in case
+        # of error(in this case, orphan VM).
+        # Explicity providing empty list of volumes to avoid misleading output.
```

Comment:
How do you differentiate between genuine error versus VM not member of any tenant?

Please add link to issue.